analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/CookieShouldBeHttpOnly.cs

```diff
@@ -50,12 +54,40 @@ public sealed class CookieShouldBeHttpOnly : ObjectShouldBeInitializedCorrectlyB
 
         internal CookieShouldBeHttpOnly(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration, DiagnosticId, MessageFormat) { }
 
+        protected override Func<bool> DetermineIfDefaultConstructorIsSafe(SonarAnalysisContext context, AnalyzerOptions options)
+        {
+            foreach (var fullPath in context.ProjectConfiguration(options).FilesToAnalyze.FindFiles("web.config"))
+            {
+                var webConfig = File.ReadAllText(fullPath);
+                if (webConfig.Contains("<system.web>") && XmlHelper.ParseXDocument(webConfig) is { } doc
+                    && IsHttpOnlyCookies(doc))
+                {
+                    return () => true;
+                }
+            }
+
+            return () => false;
+        }
+
         protected override void Initialize(TrackerInput input)
         {
             var t = CSharpFacade.Instance.Tracker.ObjectCreation;
             t.Track(input,
                 t.MatchConstructor(KnownType.Nancy_Cookies_NancyCookie),
                 t.ExceptWhen(t.ArgumentIsBoolConstant("httpOnly", true)));
         }
+
+        private static bool IsHttpOnlyCookies(XDocument document)
+        {
+            foreach (var pages in document.XPathSelectElements("configuration/system.web/httpCookies"))
+            {
+                if (pages.GetAttributeIfBoolValueIs("httpOnlyCookies", true) is { } _)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
This can be singlelined (if I haven't missed something)
```
document.XPathSelectElements("configuration/system.web/httpCookies").Any(x => x.GetAttributeIfBoolValueIs("httpOnlyCookies", true) is { } _);
```